### PR TITLE
Fix start of autocompletion

### PR DESCRIPTION
**Pull request by jvican**
There was a bug that caused the omnicompletion to start the completion one column ahead. This PR fixes this and also adds no constraints to run autocompletion when the user asks for it. For instance, now we allow it even if you don't have written any seed.

### CODE REVIEW

**Comment by jvican** [comment]
Review by @ChrisCoffey.

**Comment by jvican** [comment]
Please, LGTM as soon as you can as I want to merge this soon @ChrisCoffey :smile:

**Comment by ChrisCoffey** [comment]
taking a look now

File: ensime_shared/ensime.py [hunk 1]
```diff
@@ -833,23 +833,22 @@ def detect_row_column_start():
...
             start = col
             line = self.vim.current.line
-            # No completion from empty string
-            if start != 0 and line[start - 1] == " ":
-                start = -1
-            else:
-                while start > 0 and line[start - 1] != " ":
-                    start -= 1
-            return row, col, start + 1  # pos in no empty str
+            while start > 0 and line[start - 1] != " ":
```
Comment (ChrisCoffey) [inline ensime_shared/ensime.py]: I forgot another common terminal case, the `.` operator. I think it probably makes sense for this condition to be:

```python
while start > 0 and (line[start -1] != " " and line[start - 1] != "."):
```

So we don't overwrite symbol names when chaining methods.

**Comment by ChrisCoffey** [comment]
Other than my one comment this looks great

**Pull request merged**